src/main/js/ephox/imagetools/util/ImageResult.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * ImageResult.js
+ *
+ * Released under LGPL License.
+ * Copyright (c) 1999-2015 Ephox Corp. All rights reserved
+ *
+ * License: http://www.tinymce.com/license
+ * Contributing: http://www.tinymce.com/contributing
+ */
+
+define("ephox/imagetools/util/ImageResult", [], function() {
+
+    /**
+     * Creates data structure that will hold image data simultaneously as blob and as dataUri
+     *
+     * @method create
+     * @static
+     * @param {Blob} blob
+     * @param {String) dataUri
+     */
+    function create(blob, uri) {
+        return {
+            blob: function() {
+                return blob;
+            },
+
+            dataUri: function() {
+                return uri;
+            },
+
+            base64: function() {
+                return uri.split(',')[1];
```

Comment:
If url isn't changing, it probably makes sense to do this `split` once on create, rather than every time base64 is requested.